### PR TITLE
kubectl editor add doc.go

### LINKED ISSUE #99: Document pkg/kubectl/cmd/util/editor package
Document the [pkg/kubectl/cmd/util/editor](https://github.com/kubernetes/kubernetes/tree/master/pkg/kubectl/cmd/util/editor) package

- Add a `doc.go` with overview of the package and examples for usage
- Audit and improve type and function code comments.  Improve documentation.

### PR DESCRIPTION
**Pull request by Farice4**
Signed-off-by: Yuanbin.Chen <cybing4@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes kubernetes/kubectl#99

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/cmd/util/editor/doc.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/*
+Copyright 2017 The Kubernetes Authors.
```
Comment (seans3) [inline pkg/kubectl/cmd/util/editor/doc.go]: Copyright 2018

Comment (chenyb4) [reply pkg/kubectl/cmd/util/editor/doc.go]: Done

**Comment by nikhita** [comment]
/unassign

Will let @seans3 tag the labels.

**Pull request closed without merging**